system/security/plugins/jwtSecurity/jwtEndpoint.cpp

```diff
@@ -0,0 +1,230 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include <curl/curl.h>
+#include <iostream>
+#include <openssl/sha.h>
+#include <string.h>
+
+#include "jlog.hpp"
+
+#include "nlohmann/json.hpp"
+
+#include "jwtEndpoint.hpp"
+
+struct MemoryStruct
```

Comment:
It looks like a MemoryBuffer would provide the functionality you need and simplify the code.

Reply:
Swapped in MemoryBuffer.  Good call.